kotlin/iwa-sample-app/src/main/java/com/esri/android/samples/iwasampleapp/MainActivity.kt

```diff
@@ -0,0 +1,173 @@
+package com.esri.android.samples.iwasampleapp
+
+import android.app.Activity
+import android.os.Bundle
+import android.support.v7.app.AlertDialog
+import android.support.v7.app.AppCompatActivity
+import android.text.Editable
+import android.text.TextWatcher
+import android.widget.Toast
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.loadable.Loadable
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.security.*
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.credential_dialog.view.*
+import java.util.concurrent.CountDownLatch
+
+/**
+ * This sample app shows how to use a custom authentication challenge handler to work with
+ * IWA portals. 
+ * 
+ * 1. Enter url of the portal you want to connect in the edit text box. The url of a default test portal has
+ * been set to the box.
+ * 2. Tap on the "Sign In" button to connect to the portal as a named user. A dialog will be popped up for
+ * username/password. When a correct credential is passed, portal will be loaded. If you keep passing in
+ * wrong credential, no more prompt for credential after 5 attempts, which is actually 4 times due to a known
+ * issue in our SDK. Then portal will fail to load. You can click the "Cancel" button to cancel the sign-in
+ * process. Portal will be failed to load with different error message.
+ */
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var map: ArcGISMap
+    
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // create a Streets BaseMap
+        map = ArcGISMap(Basemap.createStreets())
+        // set the map to be displayed in this view
+        mapView.map = map
+        
+        // Set authentication challenge handler
+        AuthenticationManager.setAuthenticationChallengeHandler(IWACustomChallengeHandler(this))
+        
+        // Sign in a portal
+        signinButton.setOnClickListener{
+            // Validate portal url
+            val url : String ? = portalUrl.text.toString()
+            if (url.isNullOrEmpty()) {
+                Toast.makeText(applicationContext, "Portal url is empty. Please enter portal url!", Toast.LENGTH_SHORT).show()
+                return@setOnClickListener
+            }
+            
+            // Clear credential cache. This is for test purpose. In a real app credential cache provides
+            // better user experience.
+            AuthenticationManager.CredentialCache.clear()
+            // Create a Portal object with loginRequired as true, which forces user to sign in
+            val portal : Portal = Portal(url, true)
```

Comment:
The portal instance should be held on to in a Kotlin property:

```
var portal : Portal? = null
```

Reply:
pupated